source/extensions/filters/listener/proxy_protocol/proxy_protocol.cc

```diff
@@ -443,6 +456,24 @@ ReadOrParseState Filter::readProxyHeader(Network::ListenerFilterBuffer& buffer)
   } else {
     // continue searching buffer from where we left off
     for (; search_index_ < raw_slice.len_; search_index_++) {
+
+      if (config_.get()->allowRequestsWithoutProxyProtocol()) {
+        if (search_index_ < PROXY_PROTO_V1_SIGNATURE_LEN &&
+            buf[search_index_] != PROXY_PROTO_V1_SIGNATURE[search_index_]) {
+          possibly_v1_ = false;
+        }
+        if (search_index_ < PROXY_PROTO_V2_SIGNATURE_LEN &&
+            buf[search_index_] != PROXY_PROTO_V2_SIGNATURE[search_index_]) {
+          possibly_v2_ = false;
+        }
+        if (!possibly_v1_ && !possibly_v2_) {
```

Comment:
Thinking of the case a header with the v1 first, then follow with matched v2 header bytes. Something like:
PROXY_PROTO_V1_SIGNATURE + PROXY_PROTO_V2_SIGNATURE[sizeof(PROXY_PROTO_V1_SIGNATURE)]

Then it will skip this check.

But as we said this used in a trusted source https://github.com/envoyproxy/envoy/pull/18951/files#diff-77045fcec1762c40ff5cf70a8e2acff7bacd8a1cb27317aba0f03d5e42a3abf3R50

not sure whether we should check such strict.

I think I'm fine with this case, but If you want to implement this case, it will be great to have a test case for this.

Reply:
I thought we agreed to only look at the signatures to determine forwarding behavior? I added tests for partial v1 and partial v2 signatures; I did not think we wanted to look beyond the signature to determine forwarding behavior

Reply:
This logic is dated and has been updated regardless, please let me know if you still have any concerns with the new implementation